common/djangoapps/entitlements/rest_api/v1/tests/test_views.py

```diff
@@ -672,6 +672,34 @@ def test_get_expired_entitlement_by_uuid(self):
         results = response.data
         assert results.get('expired_at')
 
+    def test_entitlements_filter_on_course_uuid(self):
+        """
+        Verify that courses filtered properly on list of course_uuids.
+        """
+        entitlements = CourseEntitlementFactory.create_batch(5, user=self.user)
+        course_uuids_to_filter_on = {
+            str(entitlement.course_uuid)
+            for entitlement in entitlements[0:3]
+        }
+        # Create a 2nd entitlement for one of the ones to filter on
+        CourseEntitlementFactory.create(user=self.user, course_uuid=str(entitlements[0].course_uuid))
```

Comment:
I did not get it. Could you please elaborate on why are we creating another entry with the same course UUID?

Reply:
I wanted the test set to seem more realistic, because in the wild we can potentially encounter a response that has entitlements with the same course_uuid